src/zarr/storage/_local.py

```diff
@@ -208,6 +208,13 @@ async def delete(self, key: str) -> None:
         else:
             await asyncio.to_thread(path.unlink, True)  # Q: we may want to raise if path is missing
 
+    async def delete_dir(self, prefix: str) -> None:
+        # docstring inherited
+        self._check_writable()
+        path = self.root / prefix
+        if path.is_dir():
+            shutil.rmtree(path)
```

Comment:
Am I right in thinking this will just silently do nothing if a file is provided to this function? In that case we should either:

1. Document this behaviour, or 
2. Do something else like raise a warning or error

Is there a precedent from other stores which option we should take?

Reply:
Nice catch. Now it raises an error if a file is passed and does nothing if passed a non-existent path.